app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -28,45 +32,59 @@
      */
     public BlockController(Context _context) {
         this.context = _context;
-        this.user = new User();
         this.databaseHandler = new DatabaseHandler(this.context);
     }
 
+    public boolean blockExists(String owner, String key, boolean revoked) {
+        List<Block> blocks = databaseHandler.getAllBlocks(owner);
+
+        for (Block block : blocks) {
+            if (Objects.equals(block.getOwner(), owner) && Objects.equals(block.getPublicKey(), key)
+                && block.isRevoked() == revoked)
+                return true;
+        }
+
+        return false;
+    }
+
     /**
-     * adding a block to the blockchain
+     * Add a block to the database with checking if the (owner,pubkey) pair
+     * is already added to the database
      *
      * @param block Block you want to add
-     * @return returns the block you added
      */
-    public List<Block> addBlock(Block block) {
-        // Check if the block already exists
-        String owner = block.getOwner();
-        Block latest = databaseHandler.getLatestBlock(owner);
+    public void addBlock(Block block) {
 
-        if (latest == null) {
-            databaseHandler.addBlock(block);
-        } else if (latest.isRevoked()) {
-            throw new RuntimeException("Error - Block is already revoked");
-        } else {
-            if (block.isRevoked()) {
-                revokedTrustValue(latest);
-                databaseHandler.updateBlock(latest);
-                databaseHandler.addBlock(block);
-            }
-            else throw new RuntimeException("Error - Block already exists");
-        }
+        if (blockExists(block.getOwner(), block.getPublicKey(), block.isRevoked()))
+            throw new RuntimeException("block already exists");
 
-        return getBlocks(owner);
+        databaseHandler.addBlock(block);
+    }
+
+    /**
+     * Clears all blocks from the database
+     */
+    public void clearAllBlocks() {
+        databaseHandler.clearAllBlocks();
     }
 
```

Comment:
yes when you are connected to the person, his whole chain is added into the database (not the chain yet!). After you click op add contact, a block with that persons pub key and iban will be added to your chain. His whole chain will still be in your database, but his friend name will be displayed as "OwnerName"'s friend #...